### PR TITLE
[web] remove unnecessary awaits from flutter.js

### PR DESCRIPTION
**Pull request by p-mazhnik**
Fixed types for `_getNewServiceWorker` and `_waitForServiceWorkerActivation` functions.  
These functions currently expect a Promise as an argument, but we're actually passing in an already resolved value:
```js
.then(this._getNewServiceWorker)
.then(this._waitForServiceWorkerActivation);
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Review by christopherfujino — COMMENTED** [review]
~This looks correct to me. @ditman can you think of a good way to test this?~ NVM, this already has a test exemption.

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js [hunk 1]
```diff
@@ -156,17 +156,15 @@ _flutter.loader = null;
...
      *
-     * @param {Promise<ServiceWorkerRegistration>} serviceWorkerRegistrationPromise
-     * @returns {Promise<ServiceWorker>}
+     * @param {ServiceWorkerRegistration} reg
+     * @returns {ServiceWorker}
      */
-    async _getNewServiceWorker(serviceWorkerRegistrationPromise) {
-      const reg = await serviceWorkerRegistrationPromise;
-
+    _getNewServiceWorker(reg) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: Can you substitute a more descriptive name

```suggestion
    _getNewServiceWorker(serviceWorkerRegistration) {
```

Comment (p-mazhnik) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: Yeah, I just didn't want to rename existing usages of the `reg` variable below

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: please do

Comment (p-mazhnik) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: Should I also rename `serviceWorker` to the `latestServiceWorker` in  `_waitForServiceWorkerActivation` function?    
I don't like that this function currently has meaningless `const serviceWorker = latestServiceWorker;` line

Comment (p-mazhnik) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: Or it can be the other way, `latestServiceWorker` -> `serviceWorker`

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:167]: oh yeah, we don't need that extra assignment now that we don't have an await.

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js [hunk 2]
```diff
@@ -156,45 +156,40 @@ _flutter.loader = null;
...
      * state to "activated".
      *
-     * @param {Promise<ServiceWorker>} latestServiceWorkerPromise
+     * @param {ServiceWorker} serviceWorker
      * @returns {Promise<void>}
      */
-    async _waitForServiceWorkerActivation(latestServiceWorkerPromise) {
-      const serviceWorker = await latestServiceWorkerPromise;
-
+    _waitForServiceWorkerActivation(serviceWorker) {
```
Comment (ditman) [inline packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:192]: Please, keep this function `async`.

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js [hunk 3]
```diff
@@ -156,45 +156,40 @@ _flutter.loader = null;
...
       if (!serviceWorker || serviceWorker.state == "activated") {
         if (!serviceWorker) {
           return Promise.reject(
```
Comment (ditman) [inline packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:195]: (I'd prefer that this directly `throw new Error("Cannot activate a null service worker!")` here and `return;` in line 205 than making the function non async :)

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js [hunk 4]
```diff
@@ -156,45 +156,40 @@ _flutter.loader = null;
...
-      if (!reg.active && (reg.installing || reg.waiting)) {
+    async _getNewServiceWorker(serviceWorkerRegistration) {
+      if (!serviceWorkerRegistration.active && (serviceWorkerRegistration.installing || serviceWorkerRegistration.waiting)) {
         // No active web worker and we have installed or are installing
         // one for the first time. Simply wait for it to activate.
         console.debug("Installing/Activating first service worker.");
-        return reg.installing || reg.waiting;
-      } else if (!reg.active.scriptURL.endsWith(serviceWorkerVersion)) {
+        return serviceWorkerRegistration.installing || serviceWorkerRegistration.waiting;
+      } else if (!serviceWorkerRegistration.active.scriptURL.endsWith(serviceWorkerVersion)) {
```
Comment (ditman) [inline packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:173]: I think this regresses your own fix of `serviceWorkerVersion` not being a global?

**Review by ditman — COMMENTED** [review]
Thanks for fine-tuning this! However this will conflict with your other PR where you fix the global `serviceWorkerVersion` variable :)

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js (hunk 4 above)
Comment (p-mazhnik) [reply packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:173]: no, I just renamed variables here, `serviceWorkerVersion` is still the same, from the global context

File: packages/flutter_tools/lib/src/web/file_generators/js/flutter.js [hunk 5]
```diff
@@ -156,53 +156,46 @@ _flutter.loader = null;
...
         if (!serviceWorker) {
-          return Promise.reject(
-            new Error("Cannot activate a null service worker!")
-          );
+          throw new Error("Cannot activate a null service worker!");
         } else {
           console.debug("Service worker already active.");
-          return Promise.resolve();
+          return;
         }
```
Comment (ditman) [inline packages/flutter_tools/lib/src/web/file_generators/js/flutter.js:199]: Ah yes, much better! Thanks!

**Review by ditman — APPROVED** [review]
LGTM, thanks for the cleanup!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by p-mazhnik** [comment]
Blocked by https://github.com/flutter/flutter/pull/130206

**Comment by ditman** [comment]
@p-mazhnik this is independent of https://github.com/flutter/flutter/pull/130206. How about we get this one landed, and then have 130206 updated on top of this change?

**Comment by p-mazhnik** [comment]
@ditman sounds good, let's do this way!

**Comment by ditman** [comment]
OK @p-mazhnik this has immediately landed, you should be able of rebasing #130206 whenever it's convenient to you!

**Pull request merged**